wolfcrypt/src/aes.c

```diff
@@ -11438,6 +11438,83 @@ int wc_AesXtsDecrypt(XtsAes* xaes, byte* out, const byte* in, word32 sz,
         return AesXtsDecrypt_sw(xaes, out, in, sz, i);
     }
 }
+
+/* Same as wc_AesXtsEncryptSector but the sector gets incremented by the
+ * sectorSz every 32 blocks
+ *
+ * xaes     AES keys to use for block encrypt
+ * out      output buffer to hold cipher text
+ * in       input plain text buffer to encrypt
+ * sz       size of both out and in buffers
+ * sector   value to use for tweak
+ * sectorSz size of the sector
+ *
+ * returns 0 on success
+ */
+int wc_AesXtsEncryptConsecutiveSectors(XtsAes* aes, byte* out, const byte* in,
+        word32 sz, word64 sector, word32 sectorSz)
+{
+    int ret  = 0;
+    int iter = 0;
+    int sectionSz = 2 * AES_BLOCK_SIZE * AES_BLOCK_SIZE;
+    int sections  = sz / sectionSz;
+    int remainder = sz % sectionSz;
+
+    while (sections) {
+        ret = wc_AesXtsEncryptSector(aes, out + (iter * sectionSz),
+                in + (iter * sectionSz), sectionSz, sector);
+        if (ret != 0)
+            break;
+
+        sections--;
+        iter++;
+        sector += sectorSz;
+    }
+
+    if (remainder && ret == 0)
+        ret = wc_AesXtsEncryptSector(aes, out + (iter * sectionSz),
+                in + (iter * sectionSz), remainder, sector);
+
+    return ret;
+}
+
+/* Same as wc_AesXtsEncryptConsecutiveSectors but Aes key is AES_DECRYPTION type
+ *
+ * xaes     AES keys to use for block decrypt
+ * out      output buffer to hold cipher text
+ * in       input plain text buffer to encrypt
+ * sz       size of both out and in buffers
+ * sector   value to use for tweak
+ * sectorSz size of the sector
+ *
+ * returns 0 on success
+ */
+int wc_AesXtsDecryptConsecutiveSectors(XtsAes* aes, byte* out, const byte* in,
+        word32 sz, word64 sector, word32 sectorSz)
+{
+    int ret  = 0;
+    int iter = 0;
+    int sectionSz = 2 * AES_BLOCK_SIZE * AES_BLOCK_SIZE;
```

Comment:
> For most disks, each sector stores a fixed amount of user-accessible data, traditionally 512 bytes for hard disk drives (HDDs) and 2048 bytes for CD-ROMs and DVD-ROMs. Newer HDDs and SSDs use 4096-byte (4 KiB) sectors, which are known as the Advanced Format (AF).

From [Disk Sectors on wikipedia ](https://en.wikipedia.org/wiki/Disk_sector#:~:text=For%20most%20disks%2C%20each%20sector,the%20Advanced%20Format%20(AF).). Looks like it can differ, I'll add it in as a parameter